astropy/visualization/wcsaxes/axislabels.py

```diff
@@ -0,0 +1,126 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+
+from matplotlib.text import Text
+import matplotlib.transforms as mtransforms
+
+from .frame import RectangularFrame
+
+
+class AxisLabels(Text):
+
+    def __init__(self, frame, minpad=1, *args, **kwargs):
+        self._frame = frame
+        super(AxisLabels, self).__init__(*args, **kwargs)
+        self.set_clip_on(True)
+        self.set_visible_axes('all')
+        self.set_ha('center')
+        self.set_va('center')
+        self._minpad = minpad
+
+    def get_minpad(self, axis):
+        try:
+            return self._minpad[axis]
+        except TypeError:
+            return self._minpad
+
+    def set_visible_axes(self, visible_axes):
+        self._visible_axes = visible_axes
+
+    def get_visible_axes(self):
+        if self._visible_axes == 'all':
+            return self._frame.keys()
+        else:
+            return [x for x in self._visible_axes if x in self._frame]
+
+    def set_minpad(self, minpad):
+        self._minpad = minpad
+
+    def draw(self, renderer, bboxes, ticklabels_bbox_list, visible_ticks):
+
+        if not self.get_visible():
+            return
+
+        text_size = renderer.points_to_pixels(self.get_size())
+
+        for axis in self.get_visible_axes():
+
+            padding = text_size * self.get_minpad(axis)
+
+            # Find position of the axis label. For now we pick the mid-point
+            # along the path but in future we could allow this to be a
+            # parameter.
+            x_disp, y_disp = self._frame[axis].pixel[:, 0], self._frame[axis].pixel[:, 1]
+            d = np.hstack([0., np.cumsum(np.sqrt(np.diff(x_disp) ** 2 + np.diff(y_disp) ** 2))])
+            xcen = np.interp(d[-1] / 2., d, x_disp)
+            ycen = np.interp(d[-1] / 2., d, y_disp)
+
+            # Find segment along which the mid-point lies
+            imin = np.searchsorted(d, d[-1] / 2.) - 1
+
+            # Find normal of the axis label facing outwards on that segment
+            normal_angle = self._frame[axis].normal_angle[imin] + 180.
+
+            label_angle = (normal_angle - 90.) % 360.
+            if label_angle < 225 and label_angle > 135:
```

Comment:
Put `135 < label_angle < 225` instead of `label_angle < 225 and label_angle > 135`? (slightly simpler)